lua/scriptwrapper.lua

```diff
@@ -15,38 +15,56 @@ package.path = EngineDir .. "/lua/?.lua"
 local sandbox = require('sandbox')
 local stp = require('stacktraceplus')
 
+local isSandboxed
+
 local PrintStackTrace = function(msg)
 	return stp.stacktrace("", 2) .. "\nError message\n===============\n" .. msg .. "\n==============="
 end
 
-local TryRunSandboxed = function(fn)
-	local success, err = xpcall(function() sandbox.run(fn, {env = environment, quota = MaxUserScriptInstructions}) end, PrintStackTrace)
+local function merge(dest, source)
+	for k,v in pairs(source) do
+		dest[k] = dest[k] or v
+	end
+	return dest
+end
+
+local TryRun = function(fn)
+	if isSandboxed then
+		orig = fn
+		fn = function() sandbox.run(orig, {env = environment, quota = MaxUserScriptInstructions}) end
+	else
+		-- Merge OpenRA Lua API with system Lua API
+		setfenv(fn, merge(environment, _G))
```

Comment:
I guess the advantage is also that you could now intercept and proxy game objects. Write some debug code in lua.